### PR TITLE
Add ability to uninstall plugins

### PR DESCRIPTION
**Pull request by hsbt**
# Description:

Migrated from https://github.com/rubygems/bundler/pull/7409

/cc @fatkodima
______________

# Tasks:

- [ ] Describe the problem / feature
- [ ] Write tests
- [ ] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: bundler/lib/bundler/plugin/index.rb [hunk 1]
```diff
@@ -71,6 +71,15 @@ def register_plugin(name, path, load_paths, commands, sources, hooks)
         raise
       end
 
+      def remove_plugin(name)
+        @commands.delete_if {|_, v| v == name }
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/plugin/index.rb:75]: `@commands` looks like the same kind of structure as `@sources`, so it should be removed with the following I guess?
 
```suggestion
        @commands.delete(name)
```

File: bundler/lib/bundler/plugin/index.rb [hunk 2]
```diff
@@ -71,6 +71,15 @@ def register_plugin(name, path, load_paths, commands, sources, hooks)
         raise
       end
 
+      def remove_plugin(name)
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/plugin/index.rb:74]: I think `unregister_plugin` fits better?

File: bundler/lib/bundler/plugin.rb [hunk 3]
```diff
@@ -47,6 +47,36 @@ def install(names, options)
...
+        names.each do |name|
+          begin
+            if index.installed?(name)
+              Bundler.rm_rf(index.plugin_path(name))
+              index.remove_plugin(name)
+              Bundler.ui.info "Uninstalled plugin #{name}"
+            else
+              Bundler.ui.error "Plugin #{name} is not installed \n"
+            end
+          rescue PluginError => e
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/plugin.rb:71]: I don't think this error can actually be raised?

File: bundler/lib/bundler/plugin/index.rb (hunk 1 above)
Comment (fatkodima) [reply bundler/lib/bundler/plugin/index.rb:75]: I do not remember the exact structure of plugins index file right now, but here we should remove by matching value (as opposed to `delete` method, which deletes matching key).

**Comment by hsbt** [comment]
Ah, Sorry for the unfriendly situation. #3482 is good to continue this.

@fatkodima Thanks!

**Pull request closed without merging**